### PR TITLE
add options to handle base_url in manifest

### PR DESCRIPTION
**Pull request by soheibthriber**
Three possible case:
1) strip the base url
2) keep base url as it is
3) inject a relative path

### CODE REVIEW

File: src/filters/dmx_dash.c [hunk 1]
```diff
@@ -3616,6 +3649,11 @@ static const GF_FilterArgs DASHDmxArgs[] =
 	"- error: use MPD chaining once over or if error (MPD or segment download)", GF_PROP_UINT, "on", "off|on|error", GF_FS_ARG_HINT_ADVANCED},
 	{ OFFS(asloop), "when auto switch is enabled, iterates back and forth from highest to lowest qualities", GF_PROP_BOOL, "false", NULL, GF_FS_ARG_HINT_EXPERT},
 	{ OFFS(bsmerge), "allow merging of video bitstreams (only HEVC for now)", GF_PROP_BOOL, "true", NULL, GF_FS_ARG_HINT_EXPERT},
+	{ OFFS(keep_burl), "control BaseURL in manifest\n"
+		"- false: strip BaseURL (default)\n"
+		"- true: keep BaseURL\n"
+		"- inject: inject local relative URL before BaseURL value specified by relative_url option.", GF_PROP_STRING, "false", "false|true|inject", GF_FS_ARG_HINT_EXPERT},
```
Comment (jeanlf) [inline src/filters/dmx_dash.c:3655]: Could you please make this an enum (GF_PROP_UINT)  and not a string ?

**Review by jeanlf — CHANGES_REQUESTED** [review]

File: src/filters/dmx_dash.c [hunk 2]
```diff
@@ -3650,9 +3656,9 @@ static const GF_FilterArgs DASHDmxArgs[] =
 	{ OFFS(asloop), "when auto switch is enabled, iterates back and forth from highest to lowest qualities", GF_PROP_BOOL, "false", NULL, GF_FS_ARG_HINT_EXPERT},
 	{ OFFS(bsmerge), "allow merging of video bitstreams (only HEVC for now)", GF_PROP_BOOL, "true", NULL, GF_FS_ARG_HINT_EXPERT},
 	{ OFFS(keep_burl), "control BaseURL in manifest\n"
-		"- false: strip BaseURL (default)\n"
-		"- true: keep BaseURL\n"
-		"- inject: inject local relative URL before BaseURL value specified by relative_url option.", GF_PROP_STRING, "false", "false|true|inject", GF_FS_ARG_HINT_EXPERT},
+        "- 0: strip BaseURL (default)\n"
+        "- 1: keep BaseURL\n"
+        "- 2: inject local relative URL before BaseURL value specified by relative_url option.", GF_PROP_UINT, "0", "0|1|2", GF_FS_ARG_HINT_EXPERT},
```
Comment (jeanlf) [inline src/filters/dmx_dash.c:3661]: could you use strip, keep and inject for enum names instead of 0,1 and 2 ?

**Review by jeanlf — CHANGES_REQUESTED** [review]

File: src/filters/dmx_dash.c [hunk 3]
```diff
@@ -62,6 +62,14 @@ enum {
...
 };
 
+
+
+enum {
+    strip,
+    keep,
+    inject,
+};
+
```
Comment (jeanlf) [inline src/filters/dmx_dash.c:72]: Could you please use capital, prefixed names for the values as we do in the rest of the code, e.g. BURL_KEEP,BURL_STRIP,BURL_INJECT

**Review by jeanlf — CHANGES_REQUESTED** [review]

**Comment by rbouqueau** [comment]
What is the status on this PR?

**Review by jeanlf — APPROVED** [review]

**Comment by jeanlf** [comment]
thanks!

**Pull request merged**